packages/push/examples/median/main.rs

```diff
@@ -0,0 +1,210 @@
+        .with_scorer(scorer)
+        .into_collection_generator(population_size)
+        .sample(&mut rng);
+
+    let best = Best.select(&population, &mut rng)?;
+    println!("Best initial individual is {best}");
+
+    let umad = Umad::new(0.1, 0.1, &gene_generator);
+
+    let make_new_individual = Select::new(lexicase)
+        .then(GenomeExtractor)
+        .then(Mutate::new(umad))
+        .wrap::<GenomeScorer<_, _>>(scorer);
+
+    let mut generation = Generation::new(make_new_individual, population);
+
+    for generation_number in 0..max_generations {
+        match run_model {
+            RunModel::Serial => generation.serial_next()?,
+            RunModel::Parallel => generation.par_next()?,
+        }
+
+        let best = Best.select(generation.population(), &mut rng)?;
+        println!("Generation {generation_number:4} best is {best}");
+
+        if best.test_results.total_result.error == 0 {
+            println!("SUCCESS");
+            break;
+        }
+    }
+    Ok(())
+}
+
+fn score_genome(
+    genome: &Plushy,
+    training_cases: &Cases<Input, Output>,
+    penalty_value: i128,
+) -> TestResults<test_results::Error<i128>> {
+    let program = Vec::<PushProgram>::from(genome.clone());
+    training_cases
+        .iter()
+        .map(|&case: &Case<Input, Output>| run_case(case, &program, penalty_value))
+        .collect()
+}
+
+fn run_case(
+    Case {
+        input,
+        output: Output(expected),
+    }: Case<Input, Output>,
+    program: &[PushProgram],
+    penalty_value: i128,
+) -> i128 {
+    build_state(program, input).map_or(penalty_value, |start_state| {
+        // I don't think we're properly handling things like exceeding maximum
+        // stack size. I think the "Push way" here would be to take whatever
+        // value is on top of the relevant stack and go with it, but we instead
+        // return the penalty value.
+        start_state
+            .run_to_completion()
+            .map_or(penalty_value, |final_state| {
+                compute_error(&final_state, penalty_value, expected)
+            })
+    })
+}
+
+fn build_state(program: &[PushProgram], Input([a, b, c]): Input) -> Result<PushState, StackError> {
+    Ok(PushState::builder()
+        .with_max_stack_size(1000)
+        .with_program(program.to_vec())?
+        .with_int_input("a", a)
+        .with_int_input("b", b)
+        .with_int_input("c", c)
+        .build())
+}
+
+fn compute_error(final_state: &PushState, penalty_value: i128, expected: i64) -> i128 {
+    final_state
+        .stack::<i64>()
+        .top()
+        .map_or(penalty_value, |answer| {
+            i128::from(*answer)
+                .saturating_sub(i128::from(expected))
+                .abs()
+        })
+}
+
+fn instructions() -> impl Iterator<Item = PushInstruction> {
+    let int_instructions = IntInstruction::iter()
+        // Restore this line to remove `Min` from the instruction set.
+        // .filter(|&i| i != IntInstruction::Min)
```

Comment:
Does this make sense in median?

Reply:
It probably does. If you have both `min` and `max` then it's easier. If you don't have both, then you need some fancier stuff.